ops/charm.py

```diff
@@ -1417,6 +1422,61 @@ def config(self) -> model.ConfigData:
         """A mapping containing the charm's config and current values."""
         return self.model.config
 
+    def load_config(
+        self,
+        cls: Type[_ConfigType],
```

Comment:
Suggestion from James in the spec:

```
P = ParamSpec("P")

def load_config(
    self,
    factory: Callable[P, _ConfigType],
    *args: P.args,
    **kwargs: P.kwargs,
) -> _ConfigType:
```

Reply:
However, `P.args` would include all of the fields as well, whereas the pass-through `*args` only includes anything extra that needs to be passed in (`InitVar` in dataclasses terms).

Maybe just a plain `factory: Callable[..., _ConfigType]` would be best? That might allow something like:

```python
def safer_config(charm: ops.CharmBase, x:int, y:str) -> MyConfigClass:
    return MyConfigClass(charm=charm, x=x, y=y)

...

typed_config = load_config(safer_config, charm=self)
```

However, I don't think the extra type annotations would make it far enough that it would detect something like `charm="charm"` as an error.